Bottom: a4d63da39044ba22a9f221d0f68c3c14b395471f
Top:    faa1317e1251b40eab14c2d4bd0b8c739c26e6da
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 17:16:46 +0800

debug.patch


---

diff --git a/lib/cache.c b/lib/cache.c
index 1e05a99..23462a2 100644
--- a/lib/cache.c
+++ b/lib/cache.c
@@ -658,8 +658,10 @@ int cache_update_offset(struct mapent_cache *mc, const char *mkey, const char *k
 		return CHE_FAIL;
 
 	me = cache_lookup_distinct(mc, key);
+	if (me)
+		error(LOGOPT_ANY, "owner %p owner->key %s me %p me->key %s", owner,owner->key,me,me->key);
 	if (me && me->age == age) {
-		if (me->multi == owner)
+		if (me == owner)
 			return CHE_DUPLICATE;
 	}
